registry/storage/manifeststore.go

```diff
@@ -98,7 +101,9 @@ func (ms *manifestStore) Get(ctx context.Context, dgst digest.Digest, options ..
 		switch versioned.MediaType {
 		case schema2.MediaTypeManifest:
 			return ms.schema2Handler.Unmarshal(ctx, dgst, content)
-		case manifestlist.MediaTypeManifestList:
+		case ocischema.MediaTypeManifest:
+			return ms.ocischemaHandler.Unmarshal(ctx, dgst, content)
```

Comment:
I don't see the change to actually initialize ms.ocischemahandler in registry.go or an implementation of a ocischemaManifestHandler. Also it looks like Put() needs to be modified to handle the oscischema as well. (I hit this code when I tried uploading an OCI Image with the last version of the patch)